Mage.Sets/src/mage/cards/k/KayaSpiritsJustice.java

```diff
@@ -0,0 +1,266 @@
+        if (!target.canChoose(source.getControllerId(), source, game)) {
+            return false;
+        }
+        if (!controller.chooseTarget(outcome, exiledCards, target, source, game)) {
+            return false;
+        }
+
+        Card copyFromCard = game.getCard(target.getFirstTarget());
+        if (copyFromCard == null) {
+            return false;
+        }
+
+        game.addEffect(new CopyEffect(Duration.EndOfTurn, copyFromCard, copyToPermanent.getId()), source);
+        game.addEffect(new GainAbilityTargetEffect(FlyingAbility.getInstance(), Duration.EndOfTurn), source);
+
+        return true;
+    }
+}
+
+class KayaSpiritsJusticeExileEffect extends OneShotEffect {
+
+    KayaSpiritsJusticeExileEffect() {
+        super(Outcome.Exile);
+        staticText = "exile a card from a graveyard";
+    }
+
+    private KayaSpiritsJusticeExileEffect(final KayaSpiritsJusticeExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KayaSpiritsJusticeExileEffect copy() {
+        return new KayaSpiritsJusticeExileEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            TargetCardInGraveyard target = new TargetCardInGraveyard();
+            target.withNotTarget(true);
+            controller.choose(outcome, target, source, game);
+            Card card = game.getCard(target.getFirstTarget());
+            if (card != null) {
+                UUID exileId = CardUtil.getExileZoneId(game, source.getSourceId(), source.getSourceObjectZoneChangeCounter());
+                MageObject sourceObject = source.getSourceObject(game);
+                String exileName = sourceObject == null ? null : sourceObject.getIdName();
+                return controller.moveCardsToExile(card, source, game, true, exileId, exileName);
+            }
+        }
+        return false;
+    }
+}
+
+enum KayaSpiritsJusticeAdjuster implements TargetAdjuster {
+    instance;
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+
+        Player controller = game.getPlayer(ability.getControllerId());
+        if (controller == null) {
+            return;
+        }
+        ability.addTarget(new TargetPermanent(StaticFilters.FILTER_CONTROLLED_CREATURE));
+
+        for (UUID playerId : game.getState().getPlayersInRange(ability.getControllerId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player == null || player == controller) {
+                continue;
+            }
+            FilterPermanent filter = new FilterCreaturePermanent("creature that player controls");
+            filter.add(new ControllerIdPredicate(playerId));
+            ability.addTarget(new TargetPermanent(0, 1, filter));
```

Comment:
Use target withChooseHint(“from “ + player.getlogname) so you can hint user about current player name